pasture-core/src/containers/untyped_point.rs

```diff
@@ -0,0 +1,115 @@
+use anyhow::{bail, Context, Result};
+use crate::layout::{PointLayout, PointAttributeDefinition};
+use crate::containers::InterleavedPointView;
+use std::io::Cursor;
+
+/// A struct to create and handle points that layout can't be known to compile time.
+pub struct UntypedPoint<'layout> {
+    layout: &'layout PointLayout,
+    buffer: Vec<u8>,
+}
+
+impl<'layout> UntypedPoint<'layout> {
+    pub fn new(layout: &'layout PointLayout) -> Self {
+        Self {
+            layout,
+            buffer: vec![0; layout.size_of_point_entry() as usize],
+        }
+    }
+
+    pub fn get_attribute<'point>(
+        &'point self,
+        attribute: &PointAttributeDefinition,
+    ) -> Result<&'point [u8]> {
+        let attribute = self
+            .layout
+            .get_attribute(attribute)
+            .with_context(|| "Cannot find attribute.")?;
+        let start = attribute.offset() as usize;
+        let end = start + attribute.datatype().size() as usize;
+        if self.buffer.len() < end {
+            bail!("Buffer size to small.");
+        }
+        Ok(&self.buffer[start..end])
+    }
+
+    pub fn set_attribute(
+        &mut self,
+        attribute: &PointAttributeDefinition,
+        value: &[u8],
+    ) -> Result<()> {
+        let attribute = self
+            .layout
+            .get_attribute(attribute)
+            .with_context(|| "Cannot find attribute.")?;
+        let start = attribute.offset() as usize;
+        let end = start + attribute.datatype().size() as usize;        
+        if self.buffer.len() < end {
+            bail!("Buffer size to small.");
+        }
+        self.buffer[start..end].copy_from_slice(value);
+        Ok(())
+    }
+
+    //To handle byte manipulation.
+    pub fn get_buffer_cursor(&mut self) -> Cursor<&mut Vec<u8>> {
+        return Cursor::new(&mut self.buffer);
+    }
+
+    /// Returns the offset from an attribute of the layout.
+    /// If the attribute don't exist in the layout this function returns None.
+    pub fn get_offset_from_attribute(&self, attribute: &PointAttributeDefinition) -> Option<u64> {
+        self.layout
+            .get_attribute(attribute)
+            .map(|member| member.offset())
+    }
+
+    pub fn get_interlieved_point_view(&self) -> InterleavedPointView {
```

Comment:
think there is a wee typo here -> interleaved

(i am just here because i also want to use untyped_point :smile: )